### PR TITLE
Updating getApplicationAPI for camunda

### PR DESCRIPTION
**Pull request by guru-aot**
Updated getApplication API for camunda to get Program Year from the application

### CODE REVIEW

**Review by ann-aot — APPROVED** [review]
looks good 👍 . pls look code smells

File: sources/packages/api/src/route-controllers/application/models/application.model.ts [hunk 1]
```diff
@@ -35,6 +35,10 @@ export interface ApplicationDataDto {
    * Application dynamic data.
    */
   data: any;
+  /**
+   * Program Year is added to integrate the application in camunda workflows
+   */
+  programYear: String;
```
Comment (ann-aot) [inline sources/packages/api/src/route-controllers/application/models/application.model.ts:41]: 'string` small 's'

**Review by andrewsignori-aot — APPROVED** [review]
Looks good to me 👍

File: sources/packages/api/src/services/application/application.service.ts [hunk 2]
```diff
@@ -289,6 +289,18 @@ export class ApplicationService extends RecordDataModelService<Application> {
     return application;
   }
 
+  async getApplicationById(applicationId: number): Promise<Application> {
```
Comment (dheepak-aot) [inline sources/packages/api/src/services/application/application.service.ts:292]: Just a suggestion. We get application by id and program year but the method name describes only id. Service method name could be changed.

Comment (andrewsignori-aot) [reply sources/packages/api/src/services/application/application.service.ts:292]: IMO this method could keep the generic name as it is. If we need to get more information we can add more items to the select. I would not recommend creating a specific getApplicationX to every single select that we need unless there is a huge performance impact for a particular feature.

Comment (guru-aot) [reply sources/packages/api/src/services/application/application.service.ts:292]: Agreed @andrewsignori-aot . Actually we are not getting the application with respect to the program year. The where clause contains if the program year is active. We can have a discussion if you want.

Comment (dheepak-aot) [reply sources/packages/api/src/services/application/application.service.ts:292]: Not required @guru-aot @andrewsignori-aot. I am good.

**Review by dheepak-aot — APPROVED** [review]
LGTM

**Review by abschwenker — APPROVED** [review]
LGTM

**Pull request merged**